wordpress-mobile/WordPress-iOS
# [Modal Layout Picker] Create Layout Picker category bar UI using static data

**Fixes** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2437

## To test:

<details>
<summary><strong>To disable or enable the development version of Modal Layout Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker will be disabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Gutenberg Modal Layout Picker" to enable or disable the picker
            <img width=300 src="https://user-images.githubusercontent.com/3384451/90816133-c8b10580-e2f9-11ea-8584-6bbfd7e523e2.gif" />
</details>

When enabled, the Layout Picker should show when creating a new page from My Site and from the Page list.

Start by enabling the MLP (if needed) and by navigating to the Modal Layout Picker

### Select a Category
1. Select a category
    - **Expect** to see the sections condensed to be only the selected category filter
1. Deselect the category
    - **Expect** to see the other categories return

### Select Multiple-Categories 
1. Select a category
    - **Expect** to see the sections condensed to be only the selected category filter
1. Select a different category
    - **Expect** to see the new section added to the list
1. Deselect a category
    - **Expect** the list to update again to be only the selected categories
1. Deselect the rest of the categories
    - **Expect** the all of the categories to return

### Selecting a Layout then filter on the same category
1. Select a layout in a category such as Blog
1. Select the category in the filter bar that matches the category of the selected layout
    - **Expect** to see the section updated to only show the selected category, and for the selected layout to still be selected.
1. Deselect the category in the filter
    - **Expect** to see the other categories return, and for the selected layout to still be selected.

### Selecting a Layout then filter on the same category deselect layout
1. Select a layout in a category such as Blog
1. Select the category in the filter bar that matches the category of the selected layout
    - **Expect** to see the section updated to only show the selected category, and for the selected layout to still be selected.
1. Deselect the layout
    - **Expect** the bottom bar to revert to "Create a Blank Page"
1. Deselect the category in the filter
    - **Expect** to see the other categories return, and for no layout to be selected

### Selecting a Layout then filter on a different category
1. Select a layout in a category such as Blog
1. Select a category in the filter bar that does not match the category of the selected layout
    - **Expect** to see the section updated to only show the selected category, and for the bottom bar to remain as "Preview" and "Create Page"
1. Deselect the category in the filter
    - **Expect** to see the other categories return, and for the selected layout to still be selected.

### Selecting a Layout then filter on a different category select a different layout
1. Select a layout in a category such as Blog
1. Select a category in the filter bar that does not match the category of the selected layout
    - **Expect** to see the section updated to only show the selected category, and for the bottom bar to remain as "Preview" and "Create Page"
1. Select a different layout
1. Deselect the category in the filter
    - **Expect** to see the other categories return, and for the selected layout from step 3 to still be selected.

## Screenshots:

![CategoryDemo](https://user-images.githubusercontent.com/3384451/90816886-fb0f3280-e2fa-11ea-8a8a-840f769e42ef.gif)

## Additional Context: 
This is a continuation of the work in https://github.com/wordpress-mobile/WordPress-iOS/pull/14553 and will be followed up by other tickets as part of [Modal Layout Picker Project](https://github.com/wordpress-mobile/gutenberg-mobile/issues?q=is%3Aissue+is%3Aopen+%5BModal+Layout+Picker%5D)

## PR submission checklist:
